sushy_oem_idrac/resources/manager/manager.py

```diff
@@ -83,6 +102,90 @@ class DellManagerExtension(oem_base.OEMResourceBase):
+
+    def clear_job_queue(self, job_ids=['JID_CLEARALL']):
+
+        if job_ids is None:
+            return None
+
+        delete_job_queue_uri = '%s%s' % (self.job_service_uri, self.CLEAR_JOBS_URI)
+        for job_id in job_ids:
+            payload = {'JobID':job_id}
+
+            try:
+                delete_job_response = asynchronous.http_call(
+                        self._conn, 'post',
+                        delete_job_queue_uri,
+                        headers = self.HEADERS,
+                        data = payload,
+                        verify = False)
+                return delete_job_response
+
+            except (sushy.exceptions.ServerSideError,
+                    sushy.exceptions.BadRequestError) as exc:
+                LOG.error('Dell OEM clear job queue failed, %s', err_msg)
+                raise sushy.exceptions.ServerSideError(error=exc)
+
+    def reset_idrac(self,manager=None):
+        payload = {"ResetType":"GracefulRestart"}
+
+        try:
+            reset_job_response = asynchronous.http_call(
+                            self._conn, 'post',
+                            manager._actions.reset.target_uri,
+                            data=(payload))
+
+        except (sushy.exceptions.ServerSideError,
+                sushy.exceptions.BadRequestError) as exc:
+            LOG.error('Dell OEM reset idrac failed, Reason : %s', exc)
+            raise sushy.exceptions.ServerSideError(error=exc)
+
+        LOG.info("iDRAC has reset, Waiting for the iDRAC to become ready")
+        if(utils.wait_for_idrac_ready(self) == True):
+            LOG.info("Dell OEM iDRAC reset successfuly")
+            return reset_job_response
+        else:
+            err_msg = 'Timeout reched to become iDRAC ready'
+            LOG.error(err_msg)
+            raise sushy.exceptions.ConnectionError(error=err_msg)
+
+    def known_good_state(self,manager=None):
+        delete_job_response = self.clear_job_queue(job_ids=['JID_CLEARALL'])
+        reset_job_response = self.reset_idrac(manager=manager)
+        return [delete_job_response, reset_job_response]
+
+    def get_unfinished_jobs(self):
+        jobs_expand_uri = '%s%s' %(self.get_jobs_uri, self.JOB_EXPAND)
+        unfinished_jobs = []
+        try:
+            response = asynchronous.http_call(
+                    self._conn, 'get',
+                    jobs_expand_uri,
+                    verify = False)
+            data = response.json()
+            for job in data[u'Members']:
+                if((job[u'JobState'] == 'Scheduled') or
+                      (job[u'JobState'] == 'Running')):
+                    unfinished_jobs.append(job['Id'])
+            return unfinished_jobs
+        except (sushy.exceptions.ServerSideError,
+                sushy.exceptions.BadRequestError) as exc:
+            LOG.error('Dell OEM clear job queue failed, %s', err_msg)
+            raise sushy.exceptions.ServerSideError(error=exc)
```

Comment:
I think we should just do a "raise" in the line above.

Reply:
Addressed.